.github/workflows/run-tests.yml

```diff
@@ -1,19 +1,64 @@
 name: Run-Tests
+
 on:
   push:
     branches: [ main ]
   pull_request:
     branches: [ main ]
 
 jobs:
+  validate-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Fetch all history for all branches and tags
+
+      - name: Fetch all history
+        run: git fetch --no-tags --prune --depth=1 origin +refs/heads/*:refs/remotes/origin/*
+
+      - name: Check for changes in dependency files
+        id: check-deps
+        run: |
+          echo "Checking pyproject.toml and poetry.lock for changes..."
+          CHANGES=$(git diff --name-only origin/main origin/feature_ci_auto_deps | grep -E "^(pyproject.toml|poetry.lock)$" || true)
+          echo "Changed files: $CHANGES"
+          if [ ! -z "$CHANGES" ]; then
+            echo "Checking if requirements.txt has also been updated..."
+            if git diff --name-only origin/main origin/feature_ci_auto_deps | grep -q "requirements.txt"; then
+              echo "requirements.txt updated correctly."
+            else
+              echo "ERROR: pyproject.toml or poetry.lock has changed, but requirements.txt has not been updated."
+              echo "Please update requirements.txt by running 'make'."
+              exit 1
+            fi
+          else
+            echo "No changes in pyproject.toml or poetry.lock."
+          fi
+
+      - name: Validate requirements.txt update
+        if: steps.check-deps.outputs.poetry_updated == 'true'
+        run: |
+          REQUIREMENTS_CHANGE=$(git diff --name-only origin/${{ github.base_ref }} origin/${{ github.head_ref }} | grep -q "requirements.txt")
+          echo "Requirements.txt changed: $REQUIREMENTS_CHANGE"
+          if [ ! -z "$REQUIREMENTS_CHANGE" ]; then
+            echo "requirements.txt updated correctly."
+          else
+            echo "ERROR: pyproject.toml or poetry.lock has changed, but requirements.txt has not been updated."
+            echo "Please update requirements.txt by running 'make'."
+            exit 1
+          fi
+
   Test:
+    needs: validate-dependencies
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
           python-version: 3.11
-      - name: Cache Poetry install
+      - name: cache deps
```

Comment:
`caching local deps` may be a better name, then you can change the name of the caching job below as well.